portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,121 @@
+ * to this servlet. This servlet can then process the request using the file URL we get from
+ * Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+  String BLOB_URL_PROPERTY = DataServlet.BLOB_URL_PROPERTY;
+  String COMMENT = DataServlet.COMMENT;
+  String CONTENT_PROPERTY = DataServlet.CONTENT_PROPERTY;
+  String TIMESTAMP_PROPERTY = DataServlet.TIMESTAMP_PROPERTY;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the message entered by the user.
+    String message = request.getParameter("comment-input");
+
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "image");
+  
+    // Create an entity with received comment message
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity(COMMENT);
+    commentEntity.setProperty(CONTENT_PROPERTY, message);
+    commentEntity.setProperty(TIMESTAMP_PROPERTY,timestamp);
+    commentEntity.setProperty(BLOB_URL_PROPERTY, imageUrl);
+
+    // Used Datastore survice to store newly created comment entity
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
+    // Output some HTML that shows the data the user entered.
+    // A real codebase would probably store these in Datastore.
+    PrintWriter out = response.getWriter();
+    out.println("<p>Here's the image you uploaded:</p>");
+    out.println("<a href=\"" + imageUrl + "\">");
+    out.println("<img src=\"" + imageUrl + "\" />");
+    out.println("</a>");
+    out.println("<p>Here's the text you entered:</p>");
+    out.println(message);    
+    
+    // // Redirect back to the current page
+    // response.sendRedirect("/index.html");
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    // User submitted form without selecting a file, so we can't get a URL. (devserver)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      return null;
+    }
+
+    // We could check the validity of the file here, e.g. to make sure it's an image file
+    // https://stackoverflow.com/q/10779564/873165
```

Comment:
Add todo if you plan to do this later:
TODO: We could...

Reply:
Already checked the validity in dataservelet, no need for the TODO